src/material/core/datetime/date-selection-model.ts

```diff
@@ -128,7 +126,7 @@ export class MatSingleDateSelectionModel<D> extends MatDateSelectionModel<D | nu
 
 /**  A selection model that contains a date range. */
 @Injectable()
-export class MatRangeDateSelectionModel<D> extends MatDateSelectionModel<DateRange<D>, D> {
+export class MatRangeDateSelectionModel<D> extends MatDateSelectionModel<D, DateRange<D>> {
```

Comment:
Is it the `null` handling that made these changes necessary? What if we un-did the other changes and just made this `class MatRangeDateSelectionModel extends MatDateSelectionModel<D | null, DateRange<D | null>>`, would that have solved the issue?